src/Http/Routing/src/DependencyInjection/RoutingServiceCollectionExtensions.cs

```diff
@@ -110,12 +126,32 @@ public static IServiceCollection AddRouting(this IServiceCollection services)
     public static IServiceCollection AddRouting(
         this IServiceCollection services,
         Action<RouteOptions> configureOptions)
+    {
+        return services.AddRoutingCore(routeOptions =>
+        {
+            routeOptions.AddRegexConstraints();
+            configureOptions(routeOptions);
+        });
+    }
+
+    /// <summary>
+    /// Adds services required for routing requests. This is similar to
+    /// <see cref="AddRouting(IServiceCollection, Action{RouteOptions})" />
+    /// except that it excludes certain options that are not included by default for native AOT
+    /// scenarios.
```

Comment:
I'm not sure I'd explicitly call out "for native AOT" in the docs here. It would be better to say something along the lines of "excludes certain options that can be opted in separately, if needed."